### PR TITLE
Improve public docstrings

### PR DESCRIPTION
**Pull request by jpd-defra**
This PR adds more detail to public facing docstrings for classes, methods and functions.

### CODE REVIEW

**Comment by jpd-defra** [comment]
Fixes #214.

**Comment by jpd-defra** [comment]
Fixes #214 and #210.

I think I branched this off some initial graph work Ed and I were doing, so there are a few graph related changes to satisfy tests. Mainly commenting out an old test which is now defunct.

File: src/sds_data_model/raster.py [hunk 1]
```diff
@@ -11,7 +11,26 @@
...
+    data is provided, in which case, nearest neighbour interpolation is applied.
+
+    Attributes:
+    dataset(Dataset): An xarray Dataset containing one or more DataArrays.
+
+    Methods:
+    from_files: Read in a raster from file at 10m cell size and British National Grid extent.
+
+    Returns:
+    _DatasetWrapper: Dataset
```
Comment (jamesgyoung) [inline src/sds_data_model/raster.py]: Add indentation.
```suggestion
    Attributes:
        dataset(Dataset): An xarray Dataset containing one or more DataArrays.

    Methods:
        from_files: Read in a raster from file at 10m cell size and
            British National Grid extent.

    Returns:
        _DatasetWrapper: Dataset
```

File: src/sds_data_model/dataframe.py [hunk 2]
```diff
@@ -69,7 +71,28 @@
...
+
+    Attributes:
+        name (str): The name of the dataset.
+        data (Union[SparkDataFrame, GroupedData]): Tabular data, optionally containing
+        a geometry column.
+        metadata (Metadata, optional): Object of class `Metadata` containing descriptive
+        information relating to the dataset represented in `data`.
+        lookup (Dict, optional): ???
+        graph (Digraph, optional): Object of class `Digraph` containing nodes and edges
+        relating to the source data and transformations that have taken place.
```
Comment (jamesgyoung) [inline src/sds_data_model/dataframe.py]: Add indentation and `lookup` description.
```suggestion
        data (Union[SparkDataFrame, GroupedData]): Tabular data, optionally containing
            a geometry column.
        metadata (Metadata, optional): Object of class `Metadata` containing descriptive
            information relating to the dataset represented in `data`.
        lookup (Dict[str, Dict[Any, float]], optional): Dictionary of 
            `{column: value-map, ...}` for columns in the data. Not applied to the data. 
            Use of `categorize` will write the lookups used by the method to this
            attribute. This attribute is written to the output of `to_zarr`.
        graph (Digraph, optional): Object of class `Digraph` containing nodes and edges
            relating to the source data and transformations that have taken place.
```

**Review by jamesgyoung — CHANGES_REQUESTED** [review]
Looks good. Just a couple of indentation issues to my eye.

**Comment by jpd-defra** [comment]
Thanks @JamesKeny0n, closing and merging.

**Pull request merged**